appveyor.yml

```diff
@@ -1,96 +1,292 @@
-#         notifications           #
-#---------------------------------#
+  # enable patching of AssemblyInfo.* files
+  assembly_info:
+    patch: true
+    file: AssemblyInfo.*
+    assembly_version: "{version}"
+    assembly_file_version: "{version}"
+    assembly_informational_version: "{version}"
+  
+  #---------------------------------#
+  #       build configuration       #
+  #---------------------------------#
+  # build platform, i.e. x86, x64, Any CPU. This setting is optional.
+  platform: Any CPU
+  
+  # build Configuration, i.e. Debug, Release, etc.
+  configuration: Release
+  
+  build:
+    parallel: false                 # disable MSBuild parallel builds
+    project: RiotSharp.sln          # path to Visual Studio solution or project
+  
+    # MSBuild verbosity level
+    verbosity: minimal
+  
+  
+  # scripts to run before build
+  before_build:
+    - ps: (Get-Content c:\projects\RiotSharp\RiotSharpTest\App.config) | Foreach-Object {$_ -replace "TOURNAMENT_API_KEY", $Env:TOURNAMENT_API_KEY} | Set-Content c:\projects\RiotSharp\RiotSharpTest\App.config
+    - ps: (Get-Content c:\projects\RiotSharp\RiotSharpTest\App.config) | Foreach-Object {$_ -replace "API_KEY", $Env:API_KEY} | Set-Content c:\projects\RiotSharp\RiotSharpTest\App.config
+    - ps: nuget.exe restore c:\projects\RiotSharp\RiotSharp.sln
+  
+  #---------------------------------#
+  #       tests configuration       #
+  #---------------------------------#
+  test:
+    assemblies:
+      - RiotSharpTest.dll
+  
+  # code coverage
+  after_test:
+    - .\packages\OpenCover.4.6.519\tools\OpenCover.Console.exe -register:user -target:"c:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\IDE\MSTest.exe" -targetargs:"/noresults /noisolation /testcontainer:""c:\projects\RiotSharp\RiotSharpTest\bin\Release\RiotSharpTest.dll" -filter:"+[RiotSharp]*  -[RiotSharp]RiotSharp.*Endpoint.*" -excludebyattribute:*.ExcludeFromCodeCoverage* -hideskipped:All -output:.\RiotSharp_coverage.xml
+    - "SET PATH=C:\\Python34;C:\\Python34\\Scripts;%PATH%"
+    - pip install codecov
+    - codecov -f "RiotSharp_coverage.xml"
 
-notifications:
+  
+  deploy:
+    provider: NuGet
+    api_key:
+      secure: NUGET_API_KEY
+    skip_symbols: true
+    artifact: RiotSharp.nupkg
+  
+  
+  #---------------------------------#
+  #         notifications           #
+  #---------------------------------#
+  notifications:  
+    # Email
+    - provider: Email
+      to:
+        - benjamin.fradet@gmail.com
+      on_build_failure: true
+      on_build_success: false
+      on_build_status_changed: false
+    - provider: Webhook
+      url: https://webhooks.gitter.im/e/2668b239e8ba3b25f3ff
+      on_build_failure: true
+      on_build_success: false
+      on_build_status_changed: false
+  
+  
+  #---------------------------------#
+  #      default configuration      #
+  #---------------------------------#
+- 
+  # version format
+  version: 3.0.0-build{build}
```

Comment:
then why not 3.0.0.x instead of 3.0.0-buildx, the latter doesn't make sense to me